src/boot/efi/boot.c

```diff
@@ -1026,10 +1021,14 @@ static VOID config_defaults_load_from_file(Config *config, CHAR8 *content) {
 
         while ((line = line_get_key_value(content, (CHAR8 *)" \t", &pos, &key, &value))) {
                 if (strcmpa((CHAR8 *)"timeout", key) == 0) {
-                        _cleanup_freepool_ CHAR16 *s = NULL;
+                        if (strcmpa((CHAR8*) "force-menu", value) == 0)
+                                config->timeout_sec_config = UINT32_MAX;
+                        else {
+                                _cleanup_freepool_ CHAR16 *s = NULL;
 
-                        s = stra_to_str(value);
-                        config->timeout_sec_config = Atoi(s);
+                                s = stra_to_str(value);
+                                config->timeout_sec_config = Atoi(s);
+                        }
```

Comment:
i think it would make sense to sooner or late rmove this into a helper call `parse_timeout()`. The config_defaults_load_from_file() function is way too long already...